tools/downloader/topologies.py

```diff
@@ -0,0 +1,191 @@
+import sys
+import os
+import re
+import yaml
+import types
+import argparse
+
+from pathlib import Path
+
+import common
+
+_downloader = Path(__file__).parent / 'downloader.py'
+_precision_marker = '$precision'
+_cache = Path(os.getenv('OPENCV_OPEN_MODEL_ZOO_CACHE_DIR', 'models')).resolve()
+
+
+for _topology in common.load_topologies(argparse.Namespace(config=None)):
+    _name = _topology.name.replace('-', '_').replace('.', '_')
+    _files = _topology.files
+    _download_dir = _cache / _topology.subdirectory
+    _mo_args = {}
+    if _topology.mo_args:
+        for _arg in _topology.mo_args:
+            _tokens = _arg.split('=')
+            _mo_args[_tokens[0]] = _tokens[1] if len(_tokens) == 2 else None
+
+    # Resolve weights and text files names.
+    assert(len(_files) > 0)
```

Comment:
I wouldn't recommend asserting this. We already have some PyTorch models where the model definition is pulled implicitly via a dependency on `torchvision`; I don't think it's beyond the realm of possibility for a future model to have its weights also embedded in some dependency.

Plus, nothing seems to depend on this assertion anyway.

Reply:
Sure. Removed an assertion.